### DISCUSSION TITLE
Form ChoiceType radio buttons, access objects values in Twig

### DISCUSSION THREAD

**Discussion by frenci8** [symfony/symfony#46836] [Q&A] [answered]
Hi,

I have the following ChoiceType which I am populating with an array of Categories entity objects. via 'choices'.

```
->add('category', ChoiceType::class, [
    'label' => 'Category',
    'required' => true,
    'multiple' => false,
    'expanded' => true,
    'choices' => $options['categories'],
    'choice_value' => 'id',
    'choice_label' => function(?Categories $categories) {
        return $categories ? ucfirst($categories->getExtraDetails()) : '';
    }
])
```

and in twig I am using the following for loop to print every radio button: 

```
{% for child in form.category.children %}
   {{ form_widget(child, {
        'attr': {
            'class': 'radio-button-class'
        }
    }) }}
    <label for="{{ child.vars.id }}">
        {{ child.vars.label }}
    </label>
{% endfor %}
```

I want to add a custom class for each radio button based on the field extraDetails which is part of categories. How can I access that field?

**Comment by stof** [comment, marked as answer]
The simple solution is probably to use the `choice_attr` option, which accepts a callback receive the choice as argument and return an array of attributes to apply on the choice (similar structure that the `attr` option).

For an expanded ChoiceType, I don't think the template has access to the choice data when rendering each radio button.

**Reply by frenci8** [reply]
thank you, it worked.
